src/compose/ComposeBox.js

```diff
@@ -361,6 +386,14 @@ class ComposeBox extends PureComponent<Props, State> {
 
     return (
       <View style={this.styles.wrapper}>
+        {editMessage && (
+          <View style={this.styles.editingBanner}>
+            <Label text="Editing message" style={this.styles.editStatus} />
+            <Touchable onPress={this.cancelEdit}>
+              <Label text="Cancel" style={this.styles.editCancel} />
+            </Touchable>
+          </View>
+        )}
```

Comment:
I like this direction! Much better than changing the app bar at the top.

I think there are some tweaks I'd like to make to the UI. How about we do this as a separate followup PR? We'll want screenshots, and it'll just be easier to have that thread if it's not mixed into the same thread with code comments on the architectural changes.

This would also be great to start a thread on in #mobile with screenshots, to get feedback there.

Reply:
I was excited about this change, so I included it in this PR. But yeah, a followup PR will work.